src/Opdex.Platform.WebApi/Controllers/VaultGovernancesController.cs

```diff
@@ -200,6 +223,27 @@ public async Task<ActionResult<TransactionQuoteResponseModel>> QuoteCompleteProp
         return Ok(quote);
     }
 
+    /// <summary>Get Vault Proposal Pledges</summary>
+    /// <remarks>Retrieves vault proposal pledges.</remarks>
+    /// <param name="address">Address of the vault.</param>
+    /// <param name="proposalId">Id of the proposal in the vault.</param>
+    /// <param name="filters">Filter parameters.</param>
+    /// <param name="cancellationToken">Cancellation token.</param>
+    /// <returns>Vault proposal pledge paging results.</returns>
+    /// <response code="200">Vault proposal pledge results returned.</response>
+    /// <response code="400">The request is not valid.</response>
+    /// <response code="404">Either vault or proposal not found.</response>
+    [HttpGet("{address}/proposals/{proposalId}/pledges")]
```

Comment:
Were you thinking of two endpoints rather than having this as `{address}/pledges` and adding `proposalId` as a filter?

Same comment for votes

Reply:
Do we want to allow queries to be made across multiple vaults? Since all of our pool endpoints are market-agnostic, I'd say naturally it makes sense we allow queries across multiple vaults. Also, having the two endpoints, one for proposal votes/pledges and another for vote/pledge positions seems like extra work. So yeah I think you're right, having the one endpoint which supports all these queries is the way to go. I'm happy to amend them to support this purpose.

Reply:
I'd say for the endpoint we move it to `vault-governances/{address}/pledges` then an only check per vault adding a (public) `ProposalId` filter and a `pledger/voter` filter.

Reply:
Sorted